src/components/common/SessionFrame.vue

```diff
@@ -0,0 +1,52 @@
+<template>
+  <div class="session-header">
+    <slot name="image"></slot>
+    <a v-if="displayBack" @click="goBack">
+      <i class="material-icons session-back">arrow_back</i>
+    </a>
+    <h2 class="session-title">
+      <slot name="title"></slot>
+    </h2>
+    <p class="session-paragraph">
+      <slot name="paragraph"></slot>
+    </p>
+    <a @click="close">
+      <i class="material-icons session-close">close</i>
+    </a>
+  </div>
+</template>
+
+<script>
+export default {
+  name: `session-frame`,
+  props: {
+    previousRoute: {
+      type: String,
+      default: "welcome"
+    },
+    displayBack: {
+      type: Boolean,
+      default: true
+    },
+    nestedBack: {
+      type: Boolean,
+      default: true
+    }
+  },
+  methods: {
+    setState(value) {
+      this.$parent.$parent.$emit(`route-change`, value)
```

Comment:
hmmm this doesn't smell right 👃 

@colw what's your take?

Reply:
https://vuejs.org/v2/guide/components-edge-cases.html#Accessing-the-Parent-Component-Instance

Reply:
Fully agree. There are 2 things that I will improve on.
1. Those 3 props could become only one.
2. Close function could be passed down from SessionRouter.